lib/cluster_strategy/postgres.ex

```diff
@@ -0,0 +1,108 @@
+
+  When a node comes online, it begins to broadcast its name in a "heartbeat" message to the channel. All other nodes that receive this message attempt to connect to it.
+
+  This strategy does not check connectivity between nodes and does not disconnect them
+
+  ## Options
+
+  * `url` - The url of the database server (required)
+  * `heartbeat_interval` - The interval at which to send heartbeat messages in milliseconds (optional; default: 5_000)
+  * `channel_name` - The name of the channel to which nodes will listen and notify (optional; default: "cluster)
+  """
+  use GenServer
+
+  alias Cluster.Strategy
+  alias Cluster.Logger
+  alias Postgrex, as: P
+
+  def start_link(args), do: GenServer.start_link(__MODULE__, args)
+
+  def init([state]) do
+    if !state.config[:url] do
+      raise ArgumentError, "Missing required option :url"
+    end
+
+    opts =
+      Ecto.Repo.Supervisor.parse_url(state.config[:url])
+      |> Keyword.put_new(:parameters, application_name: "cluster_node_#{node()}")
+      |> Keyword.put_new(:auto_reconnect, true)
+
+    new_config =
+      state.config
+      |> Keyword.put_new(:heartbeat_interval, 5_000)
+      |> Keyword.put_new(:channel_name, "cluster")
+      |> Keyword.delete(:url)
+
+    meta = %{
+      opts: fn -> opts end,
+      conn: nil,
+      conn_notif: nil,
+      heartbeat_ref: make_ref()
+    }
+
+    {:ok, %{state | config: new_config, meta: meta}, {:continue, :connect}}
+  end
+
+  def handle_continue(:connect, state) do
+    with {:ok, conn} <- P.start_link(state.meta.opts.()),
+         {:ok, conn_notif} <- P.Notifications.start_link(state.meta.opts.()),
+         {_, _} <- P.Notifications.listen(conn_notif, state.config[:channel_name]) do
+      Logger.info(state.topology, "Connected to Postgres database")
+
+      meta = %{
+        state.meta
+        | conn: conn,
+          conn_notif: conn_notif,
+          heartbeat_ref: heartbeat(0)
+      }
+
+      {:noreply, put_in(state.meta, meta)}
+    else
+      reason ->
+        Logger.error(state.topology, "Failed to connect to Postgres: #{inspect(reason)}")
+        {:noreply, state}
+    end
+  end
+
+  def handle_info(:heartbeat, state) do
+    Process.cancel_timer(state.meta.heartbeat_ref)
+    P.query(state.meta.conn, "NOTIFY #{state.config[:channel_name]}, '#{node()}'", [])
+    ref = heartbeat(state.config[:heartbeat_interval])
+    {:noreply, put_in(state.meta.heartbeat_ref, ref)}
+  end
+
+  def handle_info({:notification, _, _, _, node}, state) do
+    node = String.to_atom(node)
+
+    if node != node() do
+      topology = state.topology
+      Logger.debug(topology, "Trying to connect to node: #{node}")
+
+      case Strategy.connect_nodes(topology, state.connect, state.list_nodes, [node]) do
+        :ok ->
+          Logger.debug(topology, "Connected to node: #{node}")
+
+        {:error, _} ->
+          Logger.error(topology, "Failed to connect to node: #{node}")
+      end
+    end
```

Comment:
imho probably okay. If we have a node connected to Postgres on the same topic it should really be connected to the cluster or something is really wrong. 

Larger question ... after we've connected okay, you think it's okay to let Erlang distribution disconnect nodes if there is a connectivity issue? Leave these concerns separated? 

While I've actually written a strategy for GCP I don't fully understand the edge cases with libcluster and e.g. a flapping node connection.

Reply:
> Larger question ... after we've connected okay, you think it's okay to let Erlang distribution disconnect nodes if there is a connectivity issue? Leave these concerns separated?

Yes, I think so. If a node goes down - let's say by a network issue -, Erlang distribution will disconnect that node from all the connected nodes. You can [monitor nodes](https://www.erlang.org/doc/man/erlang.html#monitor_node-2) if you want to track those disconnection events. But yeah, I think that this tracking could be in a separated module/genserver if you ever need it.

Reply:
> I have a question: what if a node is publishing its name to the channel, but you cannot connect to it after multiple attempts - let's say for a network error? Do you want to keep trying?

Yes, I believe we should keep trying. According to our workflow, all nodes that broadcast their names should be part of the same cluster. In cases of serious network disruptions, like a split between nodes, we want to avoid manually assigning nodes to the cluster after the issue is resolved